[Windows support for fluent-bit] added targets in makefile to build Windows artifacts

rawahars opened the pull request:
## Summary
In order to build Windows artifacts, we have added a separate target for the same in Makefile. We are using a cross-compiler to build Windows artifacts.

Additionally, we have added the same to Github action on PR.

To ensure that customers building their own plugins can build Windows artifacts for different architectures and using different cross-compilers, the build settings are customisable.

Please note that we have not modified the existing targets or added Windows target to `build` target. This is because in order to build Windows artifacts, we need to install cross-compiler which by default is not available on most runtimes. Therefore, adding a separate Windows target ensures backward-compatibility.

## Testing
Installed the cross-compiler on an Ubuntu machine and then ran `make windows-release`
Then tested the built plugin with fluent-bit on Windows instance to ensure that the logs are received in S3 via Firehose (as configured).

## Issue #, if available:
N/A

## Description of changes:
added a target for Windows release in makefile

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

PettitWesley commented on Makefile:
```diff
@@ -11,6 +11,10 @@
 # language governing permissions and limitations under the License.
 
+# Build settings.
+GOARCH ?= amd64
+COMPILER ?= x86_64-w64-mingw32-gcc # Cross-compiler for Windows
+
```
what does `?=` mean?

rawahars replied:
So by default it would take the assigned values but we can override them during make command invocation.
For example-
`make release` would lead to below command-
```
GOOS=windows GOARCH=amd64 CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go build -buildmode c-shared -o ./bin/firehose.dll ./
```

However if we use `GOARCH=arm64 make release` would invoke-
```
GOOS=windows GOARCH=arm64 CGO_ENABLED=1 CC=x86_64-w64-mingw32-gcc go build -buildmode c-shared -o ./bin/firehose.dll ./
```

rawahars commented:
Updated Readme file as well to include steps for Windows build.

Commit:
added target in makefile for building Windows artifacts and updated github workflow to run the same

In order to build Windows artifacts, we have added a target for the same in Makefile. We are using a cross-compiler to build Windows artifacts.

Also, we have added this target in the Github action.

To ensure that customers building their own plugins can build Windows binaries for different architectures and using different cross-compilers, these build settings are customizable.

rawahars commented on README.md:
```diff
@@ -16,9 +16,20 @@ Run `make` to build `./bin/firehose.so`. Then use with Fluent Bit:
 -o firehose \
 -p "region=us-west-2" \
--p "delivery-stream=example-stream"
+-p "delivery_stream=example-stream"
```
Had to push a new revision to rectify the README. For both Linux and Windows, the parameter is `delivery_stream` while it was mentioned as `delivery-stream` which would have resulted in plugin not being initialised.
